### PR TITLE
Platform 110

### PR DESCRIPTION
**Pull request by jamccomb92**
### Description of proposed changes
What does this pull request accomplish? Why should it be integrated?
Adds backend capabilities to handle looped form submissions. 
Also fixes linting issues in all files
### Checklist
_Put an `x` in the boxes that apply. _
- [x] My lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my changes work (if applicable)
- [ ] I have added or updated necessary documentation (if appropriate)

### Additional comments
Is there anything else you want me to know?

### CODE REVIEW

File: cloud/src/definer/crud.definer.js [hunk 1]
```diff
@@ -148,18 +148,42 @@ Parse.Cloud.define('postObjectsToClass', (request) => new Promise((resolve, reje
...
+            const originalKey = field.title.split('__loop');
+            // check if loop already used
+            if (originalKey[1] in Object.keys(loopedJson)) {
+              loopedJson[originalKey[1]][originalKey[0]] = field.answer;
+            } else {
+              loopedJson[originalKey[1]] = {};
+              loopedJson[originalKey[1]][originalKey[0]] = field.answer;
+            }
+          }
+        });
```
Comment (hopetambala) [inline cloud/src/definer/crud.definer.js:182]: I want to move his loop logic out of this `crud.definer`, could you move this logic to a folder named `_utils` and move this logic there?

File: cloud/src/definer/crud.definer.js [hunk 2]
```diff
@@ -148,18 +148,42 @@ Parse.Cloud.define('postObjectsToClass', (request) => new Promise((resolve, reje
...
                 - this contains a latitude/longitude which will post the location
     parseParentClassID - ID of the parseParentClass Object to associate the new post with
+    loopParentID - Custom Form that the looped form originates from
+    loop - bool t/f whether looped data contained in form
   ******************************************* */
 Parse.Cloud.define('postObjectsToClassWithRelation', (request) => new Promise((resolve, reject) => {
   const supplementaryForm = new Parse.Object(request.params.parseClass);
   const residentIdForm = new Parse.Object(request.params.parseParentClass);
   const userObject = new Parse.Object('_User');
+  const loopParentForm = new Parse.Object(request.params.parseClass);
```
Comment (hopetambala) [inline cloud/src/definer/crud.definer.js:158]: are `loopParentForm` and `supplementaryForm` different?

File: cloud/src/definer/crud.definer.js [hunk 3]
```diff
@@ -171,19 +195,53 @@ Parse.Cloud.define('postObjectsToClassWithRelation', (request) => new Promise((r
...
+          loop: false,
+          loopParentID: JSON.parse(JSON.stringify(mainObject)).objectId,
+        };
+        Parse.Cloud.run('postObjectsToClassWithRelation', postParams).then((result) => {
+          console.log(result); // eslint-disable-line
+        }, (error) => {
+          reject(error);
+        });
+      });
+    }
```
Comment (hopetambala) [inline cloud/src/definer/crud.definer.js:243]: this seems like it could be with the looping utils as well

**Review by hopetambala — CHANGES_REQUESTED** [review]

File: cloud/src/definer/crud.definer.js (hunk 2 above)
Comment (jamccomb92) [reply cloud/src/definer/crud.definer.js:158]: @hopetambala yes, the loopParentForm gets used when there is a loop used in the form (ie. one more loops on a form). This variable allows us to create a link between each of the form submissions when a loop is given. The main loop form can be considered the Parent like the SurveyData is the parent of supplementary forms.

**Review by hopetambala — APPROVED** [review]
Merging into `puente-demo` and testing app from there

**Pull request merged**